Add customized static cache implementation

helunwencser opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #4478
* #4491
* __->__ #4490

Differential Revision: [D60554455](https://our.internmc.facebook.com/intern/diff/D60554455)

helunwencser committed:
Update

[ghstack-poisoned]

helunwencser commented:
@helunwencser has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D60554455).

iseeyuan commented on examples/models/phi-3-mini/static_cache.py:
```diff
@@ -0,0 +1,38 @@
+from transformers import PretrainedConfig, StaticCache
+
+
+class ETStaticCache(StaticCache):
```
Could you add comments on why this custom static cache class is needed?

helunwencser replied:
of course

helunwencser committed:
Update on "Add customized static cache implementation"


Differential Revision: [D60554455](https://our.internmc.facebook.com/intern/diff/D60554455)

[ghstack-poisoned]

iseeyuan reviewed (approved):
LGTM. Thanks!
